README.md

```diff
@@ -0,0 +1,684 @@
+# UCAN Delegation Specification v1.0.0-rc.1
+
+## Editors
+
+- [Brooklyn Zelenka], [Fission]
+
+## Authors
+
+- [Brooklyn Zelenka], [Fission]
+- [Daniel Holmgren], [Bluesky]
+- [Irakli Gozalishvili], [Protocol Labs]
+- [Philipp Krüger], [Fission]
+
+## Dependencies
+
+- [DID]
+- [IPLD]
+- [UCAN]
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119].
+
+# 0. Abstract
+
+This specification describes the semantics and serialization format for [UCAN] delegation between principals. UCAN Delegation provides a cryptographically verifiable container, batched capabilities, hierarchical authority, and extensible caveats.
+  
+# 1. Introduction
+
+UCAN Delegation is a certificate capability system with runtime-extensibility, ad hoc caveats, cacheability, and focused on ease of use and interoperability. Delegations act as a proofs for [UCAN Invocation]s.
+
+Delegation provides a way to "transfer authority without transferring cryptographic keys". As an authorization system, it is more interested in "what" can be done than a list of "who can do what". For more on how Delegation fits into UCAN, please refer to the [high level spec][UCAN].
+
+# 2. Delegation (Envelope)
+
+Delegations MUST include a signature that validates against the `iss` DID. A Delegation Payload on its own MUST NOT be considered a valid Delegation.
+
+| Field   | Type        | Required | Description                                  |
+|---------|-------------|----------|----------------------------------------------|
+| `ucd`   | `&Payload`  | Yes      | The CID of the [Delegation Payload][Payload] |
```

Comment:
I don't know if we need to enforce it, but I expect to use inline IPLD a _lot_

Reply:
Can you say more about why this is a problem in js-ipld? That seems like a pretty serious limitation in such a library. Isn't the block limit in IPFS (not IPLD) something like 2MB?

Reply:
Yes recommended limit is around 2MB. 

> Can you say more about why this is a problem in js-ipld?

Before I go into JS specifics here I think fundamental language agnostic issue is what if I don't have a linked block ? When it's inline field that issue simply does not exist.

As of JS specifically, everything is built around [js-multiformats](https://github.com/multiformats/js-multiformats) library which has notion of the block and codecs that can encode / decode blocks. There is no notion of codecs that span multiple blocks nor generic structure for traversing objects that span multiple blocks. There had been several attempts to address this but nothing sticked and I think that is because it introduces asynchrony which in JS is far more consequential and invasive than other languages.

In go stack I think folks have embraced IPLD Prime which takes care of block loading behind the scenes and they can also block execution if needed so it's less of the issue there.

In Rust async also not as big of deal you can always block and wait, so I think loading blocks seems fine there, although having `Result` is probably not great there either.